### ISSUE TITLE
Icon Button does not trigger form submit event

### ISSUE THREAD

**Issue by JensRoland** [shoelace-style/webawesome-alpha#182]
### Describe the bug

`<wa-icon-button>` does not seem to trigger the form submit event. Adding `type="submit"` works for `<wa-button>` but not `<wa-icon-button>`.

### To Reproduce

Create a page with the latest WA CSS and JS and insert a simple form:

```html
<h3>Form</h3>
<form action="/delete" method="post" onsubmit="return confirm('Are you sure?');">
    <wa-icon-button name="trash-can" variant="regular" label="Delete"></wa-icon-button>
    <button type="submit">Delete</button>
</form>
```

If you click the normal 'Delete' button, the confirm prompt appears.

If you click the Icon button, nothing happens.

My workaround is to add `onclick="this.parentElement.requestSubmit()"` but it would be more intuitive to have the icon button behave the same as the button and wa-button.

### Demo

CodePen login was borked so I could not save my Pen. It should only take a moment to reproduce with the code above though.

### Screenshots
If applicable, add screenshots to help explain the bug.

### Browser / OS
 - OS: Mac OS X Sequoia
 - Browser: Chrome
 - Browser version: 132.0.6834.160

### Additional information
Provide any additional information about the bug here.

**Comment by claviska** [comment]
Icon buttons can't be used as submit buttons at this time. We're actually looking at expanding `<wa-button>` to support icons without a separate component, saving you bytes and one less API to memorize.

That said, we're not going to fix this since it would involve duplicating a lot of logic. For the moment, it may be best to use a regular `<wa-button type="submit">` and style it to look like you want until we get this merged.
